ui/app/AppLayouts/Wallet/stores/RootStore.qml

```diff
@@ -105,9 +96,35 @@ QtObject {
         return store
     }
 
+    // "walletSection" is a context property slow to lookup, so we cache it here
+    property var walletSectionInst: walletSection
+    property var totalCurrencyBalance: walletSectionInst.totalCurrencyBalance
+
+    readonly property var activityController: walletSectionInst.activityController
+    readonly property var tmpActivityController0: walletSectionInst.tmpActivityController0
+    readonly property var tmpActivityController1: walletSectionInst.tmpActivityController1
+    readonly property var activityDetailsController: walletSectionInst.activityDetailsController
+    readonly property var walletConnectController: walletSectionInst.walletConnectController
+
+    signal savedAddressAddedOrUpdated(added: bool, name: string, address: string, errorMsg: string)
+    signal savedAddressDeleted(name: string, address: string, errorMsg: string)
+
     property QtObject _d: QtObject {
         id: d
 
+        property var mainModuleInst: mainModule
```

Comment:
same here, why not readonly? Does it have anything to do with the fact that this is a context property? Is it missing at initialisation?

Reply:
Yup, mainModule is injected into the root context as `  singletonInstance.engine.setRootContextProperty("walletSection", newQVariant(self.view))` so it could be definitely readonly